lib/scan_beacon/bluez_advertiser.rb

```diff
@@ -1,60 +1,42 @@
 module ScanBeacon
-  class BlueZAdvertiser
+  class BlueZAdvertiser < GenericIndividualAdvertiser
  
-    attr_accessor :beacon, :parser, :ad, :addr
+    attr_accessor :addr
 
     def initialize(opts = {})
       @device_id = opts[:device_id] || BlueZ.devices.map {|d| d[:device_id]}[0]
       raise "No available devices" if @device_id.nil?
       BlueZ.device_up @device_id
-      @addr = @initial_addr = BlueZ.devices.find {|d| d[:device_id] == @device_id}[:addr]
-      self.beacon = opts[:beacon]
-      self.parser = opts[:parser]
-      self.parser ||= BeaconParser.default_parsers.find {|parser| parser.beacon_type == beacon.beacon_types.first}
+      addr = @initial_addr = BlueZ.devices.find {|d| d[:device_id] == @device_id}[:addr]
+      super(opts)
     end
 
-    def beacon=(value)
-      @beacon = value
-      update_ad
-    end
-
-    def parser=(value)
-      @parser = value
-      update_ad
-    end
-
-    def ad=(value)
-      @ad = value
-      BlueZ.set_advertisement_bytes @device_id, @ad
-    end
-
-    def start
+    def start(with_rotation = false)
+      addr = random_addr if with_rotation
       BlueZ.start_advertising @device_id, nil
-    end
```

Comment:
Done.  I did something slightly different, so I always call: `BlueZ.start_advertising @device_id, addr` which should be OK since the addr getter is initialized with either the original mac address or the randomized one